docs/manual/src/kotlin/gradle.md

```diff
@@ -61,7 +61,13 @@ dependencies {
 }
 ```
 
+> ## Attention
+>
+> It's highly recommend to call `AttachCurrentThread` when native threads startup, for example, using `jni-rs`'s [attach_current_thread_permanently].
```

Comment:
Actually, I think `native thread` (vs `java thread`) is specific glossary when we're talking about `jni`. And in my second sentence I'm saying `when calling from a rust thread`.

Maybe an example would be more helpful?

```rust
tokio::runtime::Builder::new_multi_thread().on_thread_start(|| {
    vm.attach_current_thread_permanently().unwrap();
})
```

Reply:
> Actually, I think native thread (vs java thread) is specific glossary when we're talking about jni.

Most people reading that doc will be looking at this through a Rust/UniFFI lens, which is why I think Ben made that suggestion. So while it's true that if they had a, say, c++ library which started threads which called back to kotlin it would apply to them too, I don't think that's useful to include here.

Through a UniFFI lens, it does seem safe to narrow the advice to Rust created threads even if the more general advice is "native threads". The point isn't to nit-pick the terminology but to ensure the reader of the doc understands exactly what it is recommending they do.

I think it would be great to include your example in the patch, probably with additional comments to imply *all* Rust threads which might make calls to the foreign bindings - ie, if a UniFFI component didn't use tokio but did create its own thread-pool which made callbacks, the same advice would apply? But if it created a thread-pool purely for internal use it would not. IOW, maybe make it clearer that your example is just one way Rust code might create threads which call back to the foreign bindings for which the advice would apply?